treesapp_assign_and_update.Rmd

```diff
@@ -72,31 +89,58 @@ do
 
 ## Update the reference package with new sequences
 
-The initial XmoA_seed reference package was pretty small with just 36 sequences (information was retrieved using `treesapp package view num_seqs -r XmoA_seed/final_outputs/XmoA_build.pkl`). Also, there were only bacterial sequences included in this version and we know there are archaeal ammonia oxidizers too. Let's see if we can expand on this seed reference package using the sequences from FunGene.
+The initial XmoA_seed reference package was pretty small with just 36 sequences (information was retrieved using `treesapp package view num_seqs -r XmoA_seed/final_outputs/XmoA_build.pkl`). Also, there were only bacterial sequences included in this version and we know there are archaeal ammonia oxidizers too.
+Let's see if we can expand on this seed reference package using the sequences from UniProt.
 
 ### Updating with publicly available sequences
 
-You have already seen many of these flags and arguments before, and they're used again here just to remind TreeSAPP that we want results fast. A couple of new ones are used, though. `treesapp update` won't take just any old FASTA file but relies on the outputs produced by `treesapp assign`. So we'll guide it to the relevant output directory with `--treesapp_output p_amoA_FunGene9.5_isolates_assign/`. From there, it will figure out what sequences were classified and what their assigned lineages were.
+You have already seen many of these flags and arguments before, and they're used again here just to remind TreeSAPP that we want results fast.
+A couple of new ones are used, though. `treesapp update` won't take just any old FASTA file but relies on the outputs produced by `treesapp assign`.
+So we'll guide it to the relevant output directory with `--treesapp_output UniProt_XmoA_assign/`.
+From there, it will figure out what sequences were classified and what their assigned lineages were.
+
+By default TreeSAPP will use the lineage that was assigned to each classified query sequence from the `treesapp assign` outputs.
+Although these sequences are from UniProt (and originally GenBank) they do not have NCBI accessions or taxonomy IDs in the header, so we must provide taxonomic lineages for each sequence in a table.
+Fortunately, we already have this information in the files `UniProt_AmoA.tab` and `UniProt_PmoA.tab`.
+
+- Create a table mapping sequence names to taxonomic lineage information for `treesapp update`.
+The first command creates a new file (`UniProt_XmoA.tsv`) with the table column names.
+The second command pipes all rows in the UniProt lineage tables except the header line by using the `tail` command.
+The `-q` parameter prevents `tail` from including the file name in the output.
+The UniProt table rows are appended to the new file by using `>>` instead of one `>`.
+
+```{bash eval = FALSE}
+echo -e "SeqID\tOrganism\tDomain\tPhylum\tClass\tOrder\tFamily\tGenus\tSpecies" >UniProt_XmoA.tsv
```

Comment:
Uniprot didn't order the headers the same for me, so I had to reorder them.
The error given by treeSAPP was hilarious
`I've seen some taxonomic domains in my time and, friend, 'uncultured methanotrophic bacterium' isn't one of them.`